### ISSUE TITLE
Bad code translation: while with nested if / return

### ISSUE THREAD

**Issue by KenanSulayman** [js2coffee/js2coffee#313]
``
while (i++)
        if (rc[i] !== token[i])
                throw false;
``

compiles to

```
throw false  if rc[i] isnt token[i]  while i++
```

and again back into something different

```
if ((function() {
  var _results;
  _results = [];
  while (i++) {
    _results.push(rc[i] !== token[i]);
  }
  return _results;
})()) {
  throw false;
}
```

**Comment by rstacruz** [comment]
Fixed in the upcoming 2.0 (js2coffee-redux). [Demo](http://js2coffee.github.io/js2coffee-redux/#try:%20while%20%28i%2B%2B%29%20if%20%28rc%5Bi%5D%20!%3D%3D%20token%5Bi%5D%29%20throw%20false%3B)

**Comment by KenanSulayman** [comment]
:y:
